### PR TITLE
Reduce gem size

### PR DESCRIPTION
**Pull request by alextwoods**
# Summary
The existing gem includes all tracked files from git, many of which are not needed at runtime.  The [current gem version](https://rubygems.org/gems/http-2/versions/0.11.0) is 1.95 MB.  This change reduces the size to 48KB by including only the library files required at runtime.

### CODE REVIEW

File: http-2.gemspec [hunk 1]
```diff
@@ -13,7 +11,7 @@ Gem::Specification.new do |spec|
   spec.license       = 'MIT'
   spec.required_ruby_version = '>=2.1.0'
 
-  spec.files         = `git ls-files`.split($INPUT_RECORD_SEPARATOR)
+  spec.files         = Dir["LICENSE", "README.md", "lib/**/*.rb"]
   spec.executables   = spec.files.grep(%r{^bin/}) { |f| File.basename(f) }
   spec.test_files    = spec.files.grep(%r{^(test|spec|features)/})
```
Comment (mullermp) [inline http-2.gemspec:16]: Can possibly just shorten this to spec too.

**Review by mullermp — APPROVED** [review]
Bundler development dependency seems redundant I think.

**Comment by mullermp** [comment]
Is there no CI?!

**Review by igrigorik — APPROVED** [review]
🤦🏻 on .git files.. thank you!

**Comment by mullermp** [comment]
Thanks! I'd like to get a refactoring change out before you release a new version.
